x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/settings_page/fleet_server_hosts_section.tsx

```diff
@@ -58,17 +59,21 @@ export const FleetServerHostsSection: React.FunctionComponent<FleetServerHostsSe
         fleetServerHosts={fleetServerHosts}
         deleteFleetServerHost={deleteFleetServerHost}
       />
-      <EuiSpacer size="s" />
-      <EuiButtonEmpty
-        iconType="plusInCircle"
-        href={getHref('settings_create_fleet_server_hosts')}
-        data-test-subj="settings.fleetServerHosts.addFleetServerHostBtn"
-      >
-        <FormattedMessage
-          id="xpack.fleet.settings.fleetServerHostCreateButtonLabel"
-          defaultMessage="Add Fleet Server"
-        />
-      </EuiButtonEmpty>
+      {authz.fleet.allSettings ? (
```

Comment:
self review: should probably be `authz.fleet.allSettings && authz.fleet.allAgents`